mcstatus/server.py

```diff
@@ -38,7 +38,7 @@ class MCServer(ABC):
 
     DEFAULT_PORT: int
 
-    def __init__(self, host: str, port: Optional[int] = None, timeout: float = 3):
+    def __init__(self, host: str, port: Optional[int] = None, timeout: float = 3) -> None:
```

Comment:
Annotations on `__init__`'s return aren't required as the type-checker can infer this. Also, this PR has no reason to make any changes in the server.py file, not sure why this is here.

```suggestion
    def __init__(self, host: str, port: Optional[int] = None, timeout: float = 3):
```

Reply:
Using mypy to check things while working on this, and I've set up my config to raise warnings whenever functions don't have type annotations. I agree there is no reason to do this.

Reply:
Yeah, we're using `flake8-annotations` to enforce type-hints, and `pyright` as the type-checker for this project, `mypy` isn't entirely compatible with pyright and there are some newer typing features it doesn't yet support which are used in the code-base, such as the `Self` type. I'd recommed switching to this setup to keep things compatible with what's getting run by our automated tests.